### PR TITLE
4: Add a new configuration option

### LINKED ISSUE #4: Lab Encrypted Storage: Support configuring TEE and Strongbox explicitly
As a library integrator,
I would like to be able to not only set the storage encryption enabled but also specify if it should prefer TEE or Strongbox,
So I can always pick the best option

Criteria:

- Update the library to allow specifying No encryption, TEE encryption, Strongbox encryption
- Explicitly remember this
- Update the readme and the sample app
- Add/Update tests as needed
- Create a new release

### PR DESCRIPTION
**Pull request by harri35**
# Issue
-https://github.com/LabMobi/LabEncryptedStorage/issues/4

### CODE REVIEW

File: labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/impl/KeyValueStorageEncryptedSharedPreferences.kt [hunk 1]
```diff
@@ -126,14 +140,19 @@ public class KeyValueStorageEncryptedSharedPreferences constructor(
...
-        // Note setRequestStrongBoxBacked is set to false for speed reasons for now.
-        // Will be configurable in the future.
         return MasterKey.Builder(appContext, STORAGE_MASTER_KEY_ALIAS)
             .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
-            .setRequestStrongBoxBacked(false)
+            .setRequestStrongBoxBacked(getFromPreferredType())
             .build()
     }
 
+    private fun getFromPreferredType(): Boolean {
```
Comment (lauriskr) [inline labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/impl/KeyValueStorageEncryptedSharedPreferences.kt]: The name could be better. Doesn't say what we get from the preferred type here

File: labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/LabEncryptedStorageManager.kt [hunk 2]
```diff
@@ -170,80 +192,72 @@ public open class LabEncryptedStorageManager(
...
          * Implementation for tester to test if the encrypted storage works on this given device.
          * Default: [SimpleEncryptedStorageDeviceCompatibilityTester].
-         * @param storageCompatibilityTester Implementation for tester to test if the encrypted storage works on this given device
+         *
+         * @param encryptionDeviceCompatibilityTester Implementation for tester to test if the encrypted storage works on this given device
          */
-        public fun storageCompatibilityTester(storageCompatibilityTester: EncryptedStorageCompatibilityTester): Builder =
-            apply { this.storageCompatibilityTester = storageCompatibilityTester }
+        public fun encryptionDeviceCompatibilityTester(
+            encryptionDeviceCompatibilityTester: EncryptedStorageCompatibilityTester
```
Comment (lauriskr) [inline labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/LabEncryptedStorageManager.kt]: The function name was changed but the interface class still uses the old name with "storage" in it. Maybe makes sense to use the same name for the interface, too?

**Review by lauriskr — APPROVED** [review]
Minor naming comments. Can be merged when addressed. Nice work

File: labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/impl/KeyValueStorageEncryptedSharedPreferences.kt (hunk 1 above)
Comment (harri35) [reply labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/impl/KeyValueStorageEncryptedSharedPreferences.kt]: Agreed, good point

File: labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/LabEncryptedStorageManager.kt (hunk 2 above)
Comment (harri35) [reply labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/LabEncryptedStorageManager.kt]: Good point

**Comment by harri35** [comment]
Discussed with Lauris - let's remove storage file prefixes also

**Pull request merged**